charts/osm/templates/osm-gateway-deployment.yaml

```diff
@@ -0,0 +1,54 @@
+{{- if .Values.OpenServiceMesh.featureFlags.enableOSMGateway }}
+---
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: osm-gateway
+  namespace: {{ include "osm.namespace" . }}
+  labels:
+    app: osm-gateway
+spec:
+  selector:
+    matchLabels:
+      app: osm-gateway
+  template:
+    metadata:
+      labels:
+        app: osm-gateway
+      name: osm-gateway
+    spec:
+      initContainers:
+        - name: osm-gateway-init
+          image: curlimages/curl
+          args:
+          - /bin/sh
+          - -c
+          - >
+            set -x;
+            while [ $(curl -sw '%{http_code}' "http://osm-controller.{{ include "osm.namespace" . }}.svc.cluster.local:9091/health/ready" -o /dev/null) -ne 200 ]; do
+              sleep 10;
+            done
+      containers:
+        - name: envoy
+          image: envoyproxy/envoy-alpine:v1.18.3
```

Comment:
Can we not use the image version from the vaules.yaml ?

Reply:
We should, my bad will fix this.